src/dune

```diff
@@ -4,7 +4,7 @@
   (public_name goblint-cil)
   (name cil)
   (wrapped false) ; this should be changed, but then module paths in goblint need to be prefixed
-  (libraries zarith findlib dynlink unix str stdlib-shims batteries.unthreaded)
```

Comment:
Are we sure this is unused? The syntactic search requires it for sure (it is listed there). But maybe it is indeed sufficient to have it there and inside the dune-project file?

Reply:
Yes, the syntactic search library uses and needs it, but the core CIL library itself does not. Initially I wanted to use `BatOption`, but this caused the `small1/longBlock` test to fail because it manually tries to compile an OCaml module (not via dune), which didn't work due to the additional batteries dependency (actually used then). And just trying to extend that command line to only depend on `BatOption` didn't really work, so I gave up and decided not to actually use the batteries dependency if it was spuriously here before.